packages/devkit/src/utils/string-change.ts

```diff
@@ -0,0 +1,140 @@
+export enum ChangeType {
+  Delete = 'DELETE',
+  Insert = 'Insert',
+}
+
+export interface StringDeletion {
+  type: ChangeType.Delete;
+  /**
+   * Place in the original text to start deleting characters
+   */
+  start: number;
+  /**
+   * Number of characters to delete
+   */
+  length: number;
+}
+
+export interface StringInsertion {
+  type: ChangeType.Insert;
+  /**
+   * Text to insert into the original text
+   */
+  text: string;
+  /**
+   * Place in the original text to insert new text
+   */
+  index: number;
+}
+
+/**
+ * A change to be made to a string
+ */
+export type StringChange = StringInsertion | StringDeletion;
+
+/**
+ * Use this to queue changes a string's original value and replay them
```

Comment:
This function doesn't queue anything (especially given that you resort the changes) and it doesn't really "replay" stuff. I'd change it as applies a list of changes to a string and describe when it is needed. The position is being moved etc.